src/test/java/org/reactome/release/common/dataretrieval/TestFileRetrieval.java

```diff
@@ -0,0 +1,276 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Random;
+
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.http.HttpEntity;
+import org.apache.http.StatusLine;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.HttpContext;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.mockito.Mock;
+import org.mockito.MockedConstruction;
+import org.mockito.MockedConstruction.Context;
+import org.mockito.MockedConstruction.MockInitializer;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/**
+ * @author sshorser
+ *
+ */
+public class TestFileRetrieval {
+
+	private static final String MESSAGE_CONTENT = "this is a test";
+	
+	@Mock
+	URI mockUri;
+	
+	@Mock
+	CloseableHttpClient mockClient;
+	
+	@Mock
+	CloseableHttpResponse mockResponse;
+
+	@Mock
+	StatusLine mockStatusLine;
+	
+	HttpEntity entity = new ByteArrayEntity(MESSAGE_CONTENT.getBytes());
+	
+	@Before
+	public void setup()
+	{
+		MockitoAnnotations.openMocks(this);
+	}
+	
+	/**
+	 * Test method for {@link org.reactome.addlinks.dataretrieval.FileRetriever#fetchData()}.
+	 * @throws Exception 
+	 */
+	@Test
+	public void testFetchData() throws Exception
+	{
+		try(MockedStatic<HttpClients> mockedStatic = Mockito.mockStatic(HttpClients.class);)
+		{
+			DataRetriever retriever = new FileRetriever();
+			//retrieve google - it should be pretty easy.
+			URI uri = new URI("http://www.google.com");
+			retriever.setDataURL(uri);
+			
+			Mockito.when(mockResponse.getEntity()).thenReturn(entity);
+			Mockito.when(mockResponse.getStatusLine()).thenReturn(mockStatusLine);
+			Mockito.when(mockClient.execute(any(HttpUriRequest.class))).thenReturn(mockResponse);
+			Mockito.when(mockClient.execute(any(HttpUriRequest.class), (HttpContext) any(HttpContext.class))).thenReturn(mockResponse);
+			
+			Mockito.when(HttpClients.createDefault()).thenReturn(mockClient);
+			
+			String dest = "/tmp/testFetchData_"+ String.valueOf((new Random()).nextInt(Integer.MAX_VALUE));
+			retriever.setFetchDestination(dest);
+			Duration age = Duration.of(5,ChronoUnit.SECONDS);
+			retriever.setMaxAge(age);
+			
+			retriever.fetchData();
+			assertTrue(Files.exists(Paths.get(dest)));
+			
+			//Sleep for two seconds, and then re-download because the file is stale
+			Thread.sleep(Duration.of(2,ChronoUnit.SECONDS).toMillis());
```

Comment:
Will this guarantee the file is stale since the max age is set to 5 seconds and then the sleep is for 2 seconds?

Reply:
Old commit wasn't checked in...